app/src/test/java/com/codesoom/assignment/controller/ProductControllerWebTest.java

```diff
@@ -0,0 +1,244 @@
+            }
+
+            @Test
+            @DisplayName("NOT_FOUND(404) 상태를 리턴합니다.")
+            void it_responses_404() throws Exception {
+                mockMvc.perform(get("/products/" + NOT_EXIST_ID))
+                        .andExpect(status().isNotFound());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("/products 로 POST 요청을 보내면")
+    class Describe_request_post_to_products_path {
+
+        @BeforeEach
+        void setUp() {
+            given(productService.createProduct(any(Product.class))).willReturn(product1);
+        }
+
+        @Test
+        @DisplayName("Created(201)과 요청에 따라 생성한 product를 json 형식으로 리턴합니다.")
+        void it_responses_created_and_product_by_json_type() throws Exception {
+            mockMvc.perform(post("/products")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(contentProduct))
+                    .andExpect(status().isCreated())
+                    .andExpect(content().string(contentProduct));
+        }
+    }
+
+
+    @Nested
+    @DisplayName("/products/{id} 로 PATCH 요청을 보내면")
+    class Describe_request_patch_to_products_id_path {
+
+        @Nested
+        @DisplayName("만약 조회하는 id의 product가 존재한다면")
+        class Context_with_exist_id {
+
+            private final Long EXIST_ID = 0L;
+
+            @BeforeEach
+            void setUp() {
+                given(productService.updateProduct(eq(EXIST_ID), any(Product.class))).willReturn(product1);
+            }
+
+            @Test
+            @DisplayName("OK(200)과 해당하는 id의 product를 주어진 요청대로 수정하고 json 형식으로 리턴합니다.")
+            void it_responses_200_and_updated_product_by_json_type() throws Exception {
+                mockMvc.perform(patch("/products/" + EXIST_ID)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(contentProduct))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string(contentProduct));
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 조회하는 id의 product가 존재하지 않는다면")
+        class Context_with_not_exist_id {
+
+            private final Long NOT_EXIST_ID = 100L;
+
+            @BeforeEach
+            void setUp() {
+                given(productService.updateProduct(eq(NOT_EXIST_ID), any(Product.class))).willThrow(new ProductNotFoundException(NOT_EXIST_ID));
+            }
+
+            @Test
+            @DisplayName("NOT_FOUND(404) 상태를 리턴합니다.")
+            void it_responses_404() throws Exception {
+                mockMvc.perform(patch("/products/" + NOT_EXIST_ID)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(contentProduct))
+                        .andExpect(status().isNotFound());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("/products/{id} 로 DELETE 요청을 보내면")
```

Comment:
```suggestion
    @DisplayName("DELETE /products/{id}")
```

간단하게 이렇게 쓰는 것도 좋을 것 같아요